packages/oidc-middleware/src/connectUtil.js

```diff
@@ -86,10 +86,16 @@ connectUtil.createLoginCallbackHandler = context => {
   const customHandler = routes.loginCallback.handler;
 
   if (!customHandler) {
-    return passport.authenticate('oidc', {
-      successReturnToOrRedirect: routes.loginCallback.afterCallback,
-      failureRedirect: routes.loginCallback.failureRedirect
-    });
+    // Passport successReturnToOrRedirect always try req.session.returnTo first if it's assigned
+    // Use successRedirect field if afterCallback url is explicitly set in config
```

Comment:
Should we change anything in the README to recommend a different practice than it currently does, or is this all internal routing that works to make the current recommendations work as intended?

Reply:
The change should be transparent to the end users. The comment is for maintainers only, since `passport` has some magic logic hide here https://github.com/jaredhanson/passport/blob/master/lib/middleware/authenticate.js#L257

Reply:
👍